wagtail_transfer/auth.py

```diff
@@ -31,3 +31,6 @@ def digest_for_source(source, message):
         message = message.encode()
 
     return hmac.new(key, message, hashlib.sha1).hexdigest()
+
+def requests_auth(source):
+    return settings.WAGTAILTRANSFER_SOURCES[source].get('BASIC_AUTH', None)
```

Comment:
Digging through old PRs... unfortunately this is now causing test failures with a KeyError on `WAGTAILTRANSFER_SOURCES[source]`.

Also, this could really do with some tests to confirm that the auth credentials are indeed showing up on the HTTP requests where appropriate.

Reply:
Hi @gasman, are you looking at merging this now? If so, I can look at rebasing and addressing the KeyError.

Regarding the test errors, is this with the wagtail-transfer test suite, or some other? I am intrigued why this has suddenly popped up since the PR hasn’t been merged.

Reply:
Hi @dkirkham - aiming to get a new release out today to address some Wagtail 5.x compatibility issues, so I suspect there won't be time to get this in. (However, I now see that there are some further deprecations introduced by 5.2 that will need attention for the 6.0 release, so there will probably be a follow-up fairly soon...)

The test failures are happening on the wagtail-transfer test suite. Is it possible that this wasn't retested with the full suite after the fix in https://github.com/wagtail/wagtail-transfer/pull/68/commits/de47c0093299a50f4d0dc64fe5b2523d279ba13f was applied, maybe? Before that point it would have been caught by the try/except. Otherwise, it may be that some other commits in the meantime have changed the test config.

Reply:
I haven’t any time to look at this for a few days, and the timezone here was against even having a quick look earlier. Since this hasn’t been merged anyway, it is probably safest to get your new release out (which it looks like you’ve done) and I’ll rebase to that.